hw/xbox/mcpx_apu.c

```diff
@@ -461,7 +465,34 @@ static uint64_t gp_read(void *opaque,
 {
     MCPXAPUState *d = opaque;
 
-    uint64_t r = d->gp.regs[addr];
+    assert(size == 4);
+
+    uint64_t r = 0;
+    switch (addr) {
+    case NV_PAPU_GPXMEM ... NV_PAPU_GPXMEM + 0xFFF*4: {
+        uint32_t xaddr = (addr - NV_PAPU_GPXMEM) / 4;
```

Comment:
Prefer shift operation over division, but compiler should automatically do that for us right?

Reply:
Yes. The generated code will not do a complicated division.